beacon-chain/state/getters.go

```diff
@@ -0,0 +1,179 @@
+package state
+
+import (
+	"github.com/gogo/protobuf/proto"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-bitfield"
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+// Clone --
+func (b *BeaconState) Clone() *pbp2p.BeaconState {
+	return proto.Clone(b.state).(*pbp2p.BeaconState)
+}
+
+// GenesisTime --
+func (b *BeaconState) GenesisTime() uint64 {
+	return b.state.GenesisTime
+}
+
+// Slot --
+func (b *BeaconState) Slot() uint64 {
+	return b.state.Slot
+}
+
+// Fork --
+func (b *BeaconState) Fork() *pbp2p.Fork {
+	return &pbp2p.Fork{
+		PreviousVersion: b.state.Fork.PreviousVersion,
+		CurrentVersion:  b.state.Fork.CurrentVersion,
+		Epoch:           b.state.Fork.Epoch,
+	}
+}
+
+// LatestBlockHeader --
+func (b *BeaconState) LatestBlockHeader() *ethpb.BeaconBlockHeader {
+	return &ethpb.BeaconBlockHeader{
+		Slot:       b.state.LatestBlockHeader.Slot,
+		ParentRoot: b.state.LatestBlockHeader.ParentRoot,
```

Comment:
Is this a safe copy of a slice? 

Same comment for all getters with byte slices.